src/query/expression/src/types/variant.rs

```diff
@@ -265,6 +267,12 @@ pub fn cast_scalar_to_variant(scalar: ScalarRef, tz: TzLUT, buf: &mut Vec<u8>) {
             buf.extend_from_slice(bytes);
             return;
         }
+        ScalarRef::Geometry(bytes) => {
+            let geometry = Ewkb(bytes.to_vec())
+                .to_json()
+                .expect("failed to decode wkb data");
+            jsonb::Value::String(geometry.into())
```

Comment:
geometry is a GeoJSON String, we can parse it to `jsonb::Value` instead of treat a string